### ISSUE TITLE
[LINUX] Segmentation fault when running some of the sample projects

### ISSUE THREAD

**Issue by guzzard** [AtomicGameEngine/AtomicGameEngine#713]
Opening Basic2D and BunnyMark example projects works fine, but they crash on play with "Segmentation fault (core dumped)". Before the crash I'm able to see some moving graphics for half a second.

Running Arch Linux. Example projects cloned from https://github.com/AtomicGameEngine/AtomicExamples

**Comment by guzzard** [comment]
I built wiuth "set(CMAKE_BUILD_TYPE Debug)" and attempted to get more information about the segfault with gdb. Strange thing is when running Basic2D or BunnyMark with the build debug flag set the editor behaves different. The segfault doesn't occur, but when stopping the player either by closing the window or using the stop button it seems like the thread handling the player doesn't exit properly, because the stop button doesn't change to a play button, and it's not possible to start the player again. When attempting to start the player again a SIGPIPE occurs as visible in the log below.

(gdb) exec-file /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor -log debug
(gdb) r
Starting program: /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Sat Mar 19 22:35:25 2016] INFO: Opened log file /home/username/.local/share/AtomicEditor/Logs/AtomicEditor.log
[New Thread 0x7fffe47b1700 (LWP 21713)]
[New Thread 0x7fffe3fb0700 (LWP 21714)]
[New Thread 0x7fffe37af700 (LWP 21715)]
[Sat Mar 19 22:35:25 2016] INFO: Created 3 worker threads
[Sat Mar 19 22:35:25 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/CoreData/
[Sat Mar 19 22:35:25 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/EditorData/
[Sat Mar 19 22:35:25 2016] DEBUG: Skipped autoload path 'Autoload' as it does not exist, check the documentation on how to set the 'resource prefix path'
[Sat Mar 19 22:35:25 2016] INFO: Set screen mode 2360x1475 windowed resizable
[Sat Mar 19 22:35:25 2016] INFO: Initialized input
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Textures/Ramp.png
[Sat Mar 19 22:35:25 2016] DEBUG: Loading temporary resource Textures/Ramp.xml
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Textures/Spot.png
[Sat Mar 19 22:35:25 2016] DEBUG: Loading temporary resource Textures/Spot.xml
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Techniques/NoTexture.xml
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource RenderPaths/Forward.xml
[Sat Mar 19 22:35:25 2016] INFO: Initialized renderer
[New Thread 0x7fffdaede700 (LWP 21716)]
[New Thread 0x7fffd66dc700 (LWP 21717)]
[Sat Mar 19 22:35:25 2016] INFO: Set audio mode 44100 Hz stereo interpolated
[Sat Mar 19 22:35:25 2016] INFO: Initialized engine
[0319/223525:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223525:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223525:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223525:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223525:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223525:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[0319/223525:ERROR:browser_main_loop.cc(203)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on deve
loping with the sandbox on.
[New Thread 0x7fffd50fc700 (LWP 21718)]
[New Thread 0x7fffd48fb700 (LWP 21720)]
[New Thread 0x7fffd40fa700 (LWP 21721)]
[New Thread 0x7fffd38f9700 (LWP 21722)]
[New Thread 0x7fffd30f8700 (LWP 21723)]
[0319/223525:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223525:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223525:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223525:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223525:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223525:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[New Thread 0x7fffd28f7700 (LWP 21724)]
[New Thread 0x7fffd20f6700 (LWP 21725)]
[New Thread 0x7fffd18f5700 (LWP 21726)]
[New Thread 0x7fffd10f4700 (LWP 21727)]
[New Thread 0x7fffd08f3700 (LWP 21728)]
[New Thread 0x7fffd00f2700 (LWP 21729)]
[New Thread 0x7fffcf8f1700 (LWP 21730)]
[New Thread 0x7fffcf0f0700 (LWP 21731)]
[New Thread 0x7fffce8ef700 (LWP 21733)]
[New Thread 0x7fffce0ee700 (LWP 21734)]
[0319/223525:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223525:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223525:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223525:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223525:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223525:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223525:ERROR:extension_system.cc(71)] Failed to parse extension manifest.
[New Thread 0x7fffc58cb700 (LWP 21736)]
[New Thread 0x7fffc4ebd700 (LWP 21737)]
Initiating Turbo Badger - version 0.1.1
[New Thread 0x7fffc46bc700 (LWP 21738)]
[New Thread 0x7fffc3ebb700 (LWP 21739)]
[Sat Mar 19 22:35:25 2016] INFO: Initialized system user interface
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource UI/DefaultStyle.xml
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Textures/UI.png
[Sat Mar 19 22:35:25 2016] DEBUG: Loading temporary resource Textures/UI.xml
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Fonts/Anonymous Pro.ttf
[Sat Mar 19 22:35:25 2016] DEBUG: Font face Anonymous Pro (11pt) has 624 glyphs
Skin loaded using 3 bitmaps.
[New Thread 0x7fffc286c700 (LWP 21743)]
[Sat Mar 19 22:35:25 2016] DEBUG: Started watching path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/CoreData/
[New Thread 0x7fffc206b700 (LWP 21744)]
[Sat Mar 19 22:35:25 2016] DEBUG: Started watching path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/EditorData/
[Sat Mar 19 22:35:25 2016] DEBUG: Window was moved to 0,0
[Sat Mar 19 22:35:25 2016] DEBUG: Window was resized to 2560x1575
[Sat Mar 19 22:35:25 2016] DEBUG: Reloading shaders
[Sat Mar 19 22:35:25 2016] DEBUG: Loading resource Shaders/GLSL/Basic.glsl
[Sat Mar 19 22:35:25 2016] DEBUG: Compiled vertex shader Basic(DIFFMAP VERTEXCOLOR)
[Sat Mar 19 22:35:25 2016] DEBUG: Compiled pixel shader Basic(DIFFMAP VERTEXCOLOR)
[Sat Mar 19 22:35:25 2016] DEBUG: Linked vertex shader Basic(DIFFMAP VERTEXCOLOR) and pixel shader Basic(DIFFMAP VERTEXCOLOR)
[New Thread 0x7fffbf99b700 (LWP 21748)]
[New Thread 0x7fffbf19a700 (LWP 21749)]
[New Thread 0x7fffbe999700 (LWP 21750)]
[New Thread 0x7fffbe198700 (LWP 21751)]
[New Thread 0x7fffbcfb8700 (LWP 21752)]
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[New Thread 0x7fffbc7b7700 (LWP 21753)]
[New Thread 0x7fffbbfb6700 (LWP 21754)]
[New Thread 0x7fffbb7b5700 (LWP 21756)]
[New Thread 0x7fffbafb4700 (LWP 21757)]
[New Thread 0x7fffba7b3700 (LWP 21758)]
[Thread 0x7fffbcfb8700 (LWP 21752) exited]
[Thread 0x7fffba7b3700 (LWP 21758) exited]
[Thread 0x7fffbc7b7700 (LWP 21753) exited]
[Thread 0x7fffbf99b700 (LWP 21748) exited]
[Thread 0x7fffbf19a700 (LWP 21749) exited]
[Thread 0x7fffbb7b5700 (LWP 21756) exited]
[Thread 0x7fffbe999700 (LWP 21750) exited]
[Thread 0x7fffbe198700 (LWP 21751) exited]
[Thread 0x7fffbbfb6700 (LWP 21754) exited]
[New Thread 0x7fffbbfb6700 (LWP 21764)]
[Sat Mar 19 22:35:29 2016] DEBUG: Started watching path /home/username/temp/AtomicExamples/Basic2D/Resources/
[Sat Mar 19 22:35:29 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D/Resources/
[Sat Mar 19 22:35:29 2016] ERROR: Could not open file /home/username/temp/AtomicExamples/Basic2D/BuildSettings.json
[Sat Mar 19 22:35:29 2016] ERROR: Could not open file /home/username/temp/AtomicExamples/Basic2D/UserPrefs.json
[New Thread 0x7fffbe198700 (LWP 21765)]
[Sat Mar 19 22:35:29 2016] DEBUG: Started watching path /home/username/temp/AtomicExamples/Basic2D/Cache/
[Sat Mar 19 22:35:29 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D/Cache/
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Scenes
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Scenes/Scene.scene
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Scripts
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Components
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Sprites
[Sat Mar 19 22:35:29 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/Basic2D/Resources/Modules
[Sat Mar 19 22:35:40 2016] INFO: Launching Broker /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor --windowposx 0 --windowposy 0 --windoww
idth 0 --windowheight 0 --resizable --fromEditorPlay --player --project "/home/username/temp/AtomicExamples/Basic2D/"
[New Thread 0x7fffbe999700 (LWP 21773)]
[Sat Mar 19 22:35:40 2016] INFO: Adding ResourcePaths: CoreData;PlayerData;/home/username/temp/AtomicExamples/Basic2D//;/home/username/temp/AtomicExamples/Basic2D//Resources;/home/gu
zzard/temp/AtomicExamples/Basic2D/;/home/username/temp/AtomicExamples/Basic2D/Cache
[Sat Mar 19 22:35:41 2016] INFO: Opened log file /home/username/.local/share/AtomicPlayer/Logs/AtomicPlayer.log
[Sat Mar 19 22:35:41 2016] INFO: Created 3 worker threads
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/CoreData/
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/PlayerData/
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D//
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D//Resources/
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D/
[Sat Mar 19 22:35:41 2016] INFO: Added resource path /home/username/temp/AtomicExamples/Basic2D/Cache/
[Sat Mar 19 22:35:41 2016] DEBUG: Skipped autoload path 'Autoload' as it does not exist, check the documentation on how to set the 'resource prefix path'
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Images/AtomicLogo32.png
[Sat Mar 19 22:35:41 2016] INFO: Set screen mode 2360x1475 windowed resizable
[Sat Mar 19 22:35:41 2016] INFO: Initialized input
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Textures/Ramp.png
[Sat Mar 19 22:35:41 2016] DEBUG: Loading temporary resource Textures/Ramp.xml
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Textures/Spot.png
[Sat Mar 19 22:35:41 2016] DEBUG: Loading temporary resource Textures/Spot.xml
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Techniques/NoTexture.xml
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource RenderPaths/Forward.xml
[Sat Mar 19 22:35:41 2016] INFO: Initialized renderer
[Sat Mar 19 22:35:41 2016] INFO: Set audio mode 44100 Hz stereo interpolated
[Sat Mar 19 22:35:41 2016] INFO: Initialized engine
[0319/223541:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223541:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223541:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223541:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223541:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223541:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[0319/223541:ERROR:browser_main_loop.cc(203)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on deve
loping with the sandbox on.
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/223541:ERROR:extension_system.cc(71)] Failed to parse extension manifest.
Initiating Turbo Badger - version 0.1.1
[Sat Mar 19 22:35:41 2016] INFO: Initialized system user interface
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource UI/DefaultStyle.xml
[0319/223541:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223541:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223541:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223541:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223541:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223541:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Textures/UI.png
[Sat Mar 19 22:35:41 2016] DEBUG: Loading temporary resource Textures/UI.xml
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Fonts/Anonymous Pro.ttf
[Sat Mar 19 22:35:41 2016] DEBUG: Font face Anonymous Pro (11pt) has 624 glyphs
[0319/223541:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/223541:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/223541:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/223541:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/223541:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/223541:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
Skin loaded using 2 bitmaps.
[Sat Mar 19 22:35:41 2016] INFO: Starting IPCWorker --ipc-server=74
[Sat Mar 19 22:35:41 2016] INFO: Starting IPCWorker --ipc-client=88
[Sat Mar 19 22:35:41 2016] ERROR: Opened IPC transport fd = 88
[Sat Mar 19 22:35:41 2016] DEBUG: Reloading shaders
[Sat Mar 19 22:35:41 2016] INFO: Loading scene from Scenes/Scene.scene
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Components/Star.js
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Sprites/star.png
[Sat Mar 19 22:35:41 2016] DEBUG: Loading resource Shaders/GLSL/Atomic2D.glsl
[Sat Mar 19 22:35:41 2016] DEBUG: Compiled vertex shader Atomic2D()
[Sat Mar 19 22:35:41 2016] DEBUG: Compiled pixel shader Atomic2D()
[Sat Mar 19 22:35:41 2016] DEBUG: Linked vertex shader Atomic2D() and pixel shader Atomic2D()
[Thread 0x7fffbafb4700 (LWP 21757) exited]

Thread 1 "AtomicEditor" received signal SIGPIPE, Broken pipe.
0x00007ffff797309d in write () from /usr/lib/libpthread.so.0
(gdb) bt full
#0  0x00007ffff797309d in write () from /usr/lib/libpthread.so.0
No symbol table info available.
#1  0x00000000015f0b7c in ?? ()
No symbol table info available.
#2  0x00003676e6c12950 in ?? ()
No symbol table info available.
#3  0x0000000000000008 in ?? ()
No symbol table info available.
#4  0x00007fffffffba00 in ?? ()
No symbol table info available.
#5  0x0000004a011b72de in ?? ()
No symbol table info available.
#6  0x00000000120bc400 in ?? ()
No symbol table info available.
#7  0x00007fffffffbac0 in ?? ()
No symbol table info available.
#8  0x0000000000000000 in ?? ()
No symbol table info available.
(gdb)

**Comment by guzzard** [comment]
output from valgrind (AE built without the debug flag)

==14449==
==14449== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==14449==  General Protection Fault
==14449==    at 0xCC7ABA: Atomic::VectorBuffer::Write(void const*, unsigned int) (in /home/guzzard/temp/AtomicGameEngineQt/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==14449==    by 0xDBCCC8: Atomic::IPCChannel::Receive() (in /home/guzzard/temp/AtomicGameEngineQt/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==14449==    by 0x12438C7: Atomic::IPCBroker::ThreadFunction() (in /home/guzzard/temp/AtomicGameEngineQt/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==14449==    by 0xC37669: Atomic::ThreadFunctionStatic(void*) (in /home/guzzard/temp/AtomicGameEngineQt/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==14449==    by 0x5096423: start_thread (in /usr/lib/libpthread-2.23.so)
==14449==    by 0xD5B9CBC: clone (in /usr/lib/libc-2.23.so)

**Comment by CTrauma** [comment]
I get the same thing:

```
#0  0x0000000000cd309a in Atomic::VectorBuffer::Write(void const*, unsigned int) ()
#1  0x0000000000dd68a1 in Atomic::IPCChannel::Receive() ()
#2  0x0000000000ddb658 in Atomic::IPCBroker::ThreadFunction() ()
#3  0x0000000000c2d46a in Atomic::ThreadFunctionStatic(void*) ()
#4  0x00007ffff7bc26aa in start_thread (arg=0x7fffc9c9c700) at pthread_create.c:333
#5  0x00007ffff0a91e9d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

the IPC stuff in Linux needs some love.

**Comment by kernelcrasher** [comment]
I get the same thing. The Basic 2D example runs for around 0.5-1 second, then crashes with a segfault.

**Comment by guzzard** [comment]
I noticed something that _might_ be helpful in investigating this issue.

If I build a **Release** version through Qt Creator or using the Build_AtomicEditor.sh script the editor crashes with this kind of error message (valgrind) when trying to play any example project:
==25650== Process terminating with default action of signal 11 (SIGSEGV): dumping core
==25650==  General Protection Fault
==25650==    at 0xBE3B5F: Atomic::VectorBuffer::Write(void const*, unsigned int) (in /home/guzzard/git/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==25650==    by 0xCDE920: Atomic::IPCChannel::Receive() (in /home/guzzard/git/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==25650==    by 0x11D8A0F: Atomic::IPCBroker::ThreadFunction() (in /home/guzzard/git/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==25650==    by 0xB511E8: Atomic::ThreadFunctionStatic(void*) (in /home/guzzard/git/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor)
==25650==    by 0x5096483: start_thread (in /usr/lib/libpthread-2.23.so)
==25650==    by 0xC7EF6DC: clone (in /usr/lib/libc-2.23.so)

However if i build a **Debug** version through Qt Creator it does not crash. I have tested many of the example projects and none of them crash when using the debug version.

So it seems to me like some optimization in the release version causes the crash somehow?

**Comment by sim-ple** [comment]
Same problem like @guzzard on Kubuntu 16.04 64Bit

**Comment by JoshEngebretson** [comment]
Closed via #982
